aim2dat/strct/strct_validation.py

```diff
@@ -57,6 +58,19 @@ def _structure_validate_positions(positions, is_cartesian, cell, inv_cell, pbc):
                 positions_cart.append(
                     tuple(float(p) for p in np.transpose(cell).dot(np.array(position)))
                 )
+    if cell is None:
+        pos2check = positions_cart
+    else:
+        pos2check = []
+        for pos in positions_scaled:
+            pos2check.append([round(pos[i], 15) % 1 if pbc[i] else pos[i] for i in range(3)])
+
+    dist_m = cdist(pos2check, pos2check)
+    for i in range(len(pos2check)):
+        for j in range(i):
+            if dist_m[i][j] < 1e-3:
+                raise ValueError(f"Sites {i} and {j} have the same position.")
```

Comment:
In this case, we would add each site pair to a list and print the pairs at the end. The idea is that we will find all sites with the same position and do not need to run the script multiple times to find all possibilities.
```suggestion
    sites = set()
    for i in range(len(pos2check)):
        for j in range(i):
            if dist_m[i][j] < 1e-3
                sites.add((i,j))
    if sites:
        raise ValueError("Sites with the same position: " + ", ".join([f"{i}" for i in sites]))
```